python/nav/report/generator.py

```diff
@@ -352,6 +358,17 @@ def _parse_single_field(self, field, value):
                     value = [None, None]
 
         self.config.where.append(field + " " + negate + operat + " %s")
+        if multi:
+            for i in range(len(value) - 1):
+                if negate:
+                    self.config.where.extend(
+                        " and " + field + " " + negate + operat + " %s"
+                    )
+                else:
+                    self.config.where.extend(
+                        " or " + field + " " + negate + operat + " %s"
+                    )
+                self.config.where = ["".join(self.config.where)]
```

Comment:
I think you need to rethink the logic here somewhat: This causes the combination logic to break down, because the AND operator logic has priority over OR.

Simple example: I search the vendor report for vendorid in `juniper,cisc*` and add the constraint that `ipcount<2`. This yields an SQL where statement like 

```sql
WHERE vendorid ilike 'juniper' or vendorid ilike 'cisc%' AND ipcount < '2'
```

The multivalue match for vendorid needs to be grouped using parentheses for the matching logic to stay unchanged, as this SQL will match everything that has either `vendorid ilike 'juniper'` OR everything that has `vendorid ilike 'cisc%'` with an ipcount of less than 2. (i.e. principally the same as 
```sql
WHERE vendorid ilike 'juniper' or (vendorid ilike 'cisc%' AND ipcount < '2')
```